tekton/release-cheatsheet.md

```diff
@@ -0,0 +1,139 @@
+
+1. `cd` to root of Results git checkout.
+
+1. Make sure the release `Task` and `Pipeline` are up-to-date on the
+   cluster.
+
+   - [release-resultss-release](https://github.com/tektoncd/results/blob/main/tekton/release.yaml)
+
+     This uses [ko](https://github.com/google/ko) to build all container images we release and generate the `release.yaml`
+     ```shell script
+     kubectl apply -f tekton/release.yaml
+     ```
+
+1. Select the commit you would like to build the release from, most likely the
+   most recent commit at https://github.com/tektoncd/results/commits/main
+   and note the commit's hash.
+
+1. Create environment variables for bash scripts in later steps.
+
+    ```bash
+    VERSION_TAG=# UPDATE THIS. Example: v0.5.0
+    RELEASE_GIT_SHA# SHA of the release to be released
+    ```
+
+1. Confirm commit SHA matches what you want to release.
+
+    ```bash
+    git show $RELEASE_GIT_SHA
+    ```
+
+1. Create a workspace template file:
+
+   ```bash
+   cat <<EOF > workspace-template.yaml
+   spec:
+     accessModes:
+     - ReadWriteOnce
+     resources:
+       requests:
+         storage: 1Gi
+   EOF
+   ```
+
+1. Execute the release pipeline.
+
+
+    ```bash
+        tkn --context dogfooding pipeline start results-release \
+        --param=revision="${RELEASE_GIT_SHA}"  \
+        --param=version="${VERSION_TAG}"  \
+        --param=bucket=gs://tekton-releases/results \
+        --workspace name=release-secret,secret=release-secret \
+        --workspace  name=ws,volumeClaimTemplateFile=workspace-template.yaml    ```
+
+1. Watch logs of result-release.
+
+1. Once the pipeline run is complete, check its results:
+
+   ```bash
+   tkn --context dogfooding pr describe <pipeline-run-name>
+
+   (...)
+   📝 Params
+
+   NAME                    VALUE
+   revision                 6ea31d92a97420d4b7af94745c45b02447ceaa19
+
+   (...)
+   ```
+
+   The `commit-sha` should match `$RELEASE_GIT_SHA`.
+   The two URLs can be opened in the browser or via `curl` to download the release manifests.
+
+    1. The YAMLs are now released! Anyone installing Tekton Results will now get the new version. Time to create a new GitHub release announcement:
+
+    1. Create additional environment variables
+
+    ```bash
+    OLD_VERSION=# Example v0.4.0
+    TEKTON_PACKAGE=tektoncd/results
+    ```
+
+
+    1. Execute the Draft Release task.
+
+    ```bash
+    tkn --context dogfooding pipeline start   \
+    --workspace name=shared,volumeClaimTemplateFile=workspace-template.yaml    \
+    --workspace name=credentials,secret=release-secret  \
+    -p package="${TEKTON_PACKAGE}"     -p git-revision="${RELEASE_GIT_SHA}"  \
+    -p release-tag="${VERSION_TAG}"     -p previous-release-tag="${OLD_VERSION}"   \
+    -p release-name="Tekton Results"     -p bucket="gs://tekton-releases/results"   \
+    -p rekor-uuid="$REKOR_UUID"  --dry-run    release-draft    ```
```

Comment:
The [release pipeline](https://github.com/tektoncd/results/blob/main/tekton/release.yaml) is does not produce the results required for chains to sign the artifacts and publish the attestation to rekor, so there will be no rekor uuid until that is implemented.

I would recommend using a publish task similar to that of [pipeline](https://github.com/tektoncd/pipeline/blob/main/tekton/publish.yaml) since it already solves the integration with chain plus other issues like:
- optionally tagging both the version as well as `latest`
- publishing images to regional instances of gcr
- check the published images with koparse
- include the source code from vendor in images to meet license requirements

Reply:
Yes, we can probably target this for v0.7.0

Reply:
@enarha I used that param because it's needed by the pipeline.  It's empty in our case but in future release, we will use that.